widget/discrepancy-note/dn-widget.js

```diff
@@ -574,12 +574,26 @@ class Comment extends Widget {
             widget.style[ o[ 0 ] ] = o[ 1 ];
         } );
 
-        // Scroll directly to Query modal if not enough height for user to view the Query modal
-        // https://github.com/OpenClinica/enketo-express-oc/issues/378
-        if ( this.linkedQuestion.classList.contains( 'or-appearance-image-map' ) ) {
-            setTimeout( () => {
-                this._scrollToview();
-            }, 500 );
+        // https://github.com/OpenClinica/enketo-express-oc/issues/495
+        const maps = document.querySelectorAll( '.or-appearance-image-map' );
+        let count = 0;
+        const dn = this;
+        if ( maps.length > 0 ) {
+            const mutationObserver = new MutationObserver( mutations => {
+                mutations.forEach( mutation => {
+                    if ( mutation.attributeName === 'viewBox' && mutation.target.tagName === 'svg' ) {
+                        count++;
+                        if ( count === maps.length ) {
+                            dn._scrollToview();
+                        }
+                    }
+                });
+            });
+
+            mutationObserver.observe( document, {
```

Comment:
Would observing `this.linkedQuestion` instead of the whole `document` work as well? More targeted.

Reply:
Otherwise, this seems like a good approach. My only concern is:

Is it possible for the SVG to be created before this tracker starts tracking (and _scrollToview never gets called). Is this dependent on the order of initialization of the dn and imagemap widgets? If so, would it be possible to get this to work regardless of the order in which widgets are loaded (or asynchronous stuff is happening)?

Reply:
Observing `this.linkedQuestion` will not catch the maps mutation if it does not contain any map. I update the code to use the `this.element.closest( 'form' )` instead of document to be more specific and added code to disconnect the observer.

> Is it possible for the SVG to be created before this tracker starts tracking (and _scrollToview never gets called)

I think that is possible, but I'm not found the case to test it yet. For this case, I think it's not an issue if `_scrollToview ` never gets called as long all the media is fully loaded before because from what I observe as default the enketo will scroll to the DN modal directly after loaded and if there are no media that still loaded the scroll position will not change.

> Is this dependent on the order of initialization of the dn and imagemap widgets?

Yes, this issue happens when we open DN on the question after or on the image map widget, especially when there is more than one image map before that DN, that'll push the DN out of the screen after they fully loaded.

> If so, would it be possible to get this to work regardless of the order in which widgets are loaded (or asynchronous stuff is happening)?

CMIIW, I think the current code will work regardless of the order, the pros are the DN will be scrolled to the correct position after the image map loaded, and the cons are for the DN that is positioned before the image map they will have a slight reposition after the image map fully loaded because the `_scrollToview` is called.

Reply:
Thanks!